src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -301,17 +310,121 @@ void FilterInvalidParameters(policy_table::RpcParameters& rpc_parameters) {
+ */
+void FilterInvalidApplicationParameters(
+    policy_table::ApplicationParams& app_params) {
+  // Filter AppHMIType array
+  policy_table::AppHMITypes valid_app_hmi_types;
+  policy_table::AppHMITypes app_hmi_types = *(app_params.AppHMIType);
+  for (const auto& app_hmi_type : app_hmi_types) {
+    if (app_hmi_type.is_valid()) {
+      valid_app_hmi_types.push_back(app_hmi_type);
+    }
+  }
+  app_params.AppHMIType->swap(valid_app_hmi_types);
+
+  // Filter RquestTypes array
+  policy_table::RequestTypes valid_request_types;
+  policy_table::RequestTypes request_types = *(app_params.RequestType);
+  for (const auto& request_type : request_types) {
+    if (request_type.is_valid()) {
+      valid_request_types.push_back(request_type);
+    }
+  }
+  if (valid_request_types.empty() && !request_types.empty()) {
+    // An empty RequestType array will allow all request types. No valid
+    // parameters are in the filtered array, so assign an uninitizlied value to
+    // for array to be "omitted"
+    policy_table::RequestTypes omitted_request_types;
+    *(app_params.RequestType) = omitted_request_types;
+  } else {
+    app_params.RequestType->swap(valid_request_types);
+  }
+
+  // Filter RquestSubTypes array
+  policy_table::RequestSubTypes valid_request_sub_types;
+  policy_table::RequestSubTypes request_sub_types =
+      *(app_params.RequestSubType);
+  for (const auto& request_sub_type : request_sub_types) {
+    if (request_sub_type.is_valid()) {
+      valid_request_sub_types.push_back(request_sub_type);
+    }
+  }
+  if (valid_request_types.empty() && !request_types.empty()) {
+    // An empty RequestSubType array will allow all request types. No valid
+    // parameters are in the filtered array, so assign an uninitizlied value to
+    // for array to be "omitted"
+    policy_table::RequestTypes omitted_request_types;
+    *(app_params.RequestType) = omitted_request_types;
+  } else {
+    app_params.RequestType->swap(valid_request_types);
+  }
+
+  // Filter moduleType array
+  policy_table::ModuleTypes valid_module_types;
+  policy_table::ModuleTypes module_types = *(app_params.moduleType);
+  for (const auto& module_type : module_types) {
+    if (module_type.is_valid()) {
+      valid_module_types.push_back(module_type);
+    }
+  }
+  if (valid_module_types.empty() && !module_types.empty()) {
+    // An empty moduleType array will allow all request types. No valid
+    // parameters are in the filtered array, so assign an uninitizlied value to
+    // for array to be "omitted"
+    policy_table::ModuleTypes omitted_module_types;
+    *(app_params.moduleType) = omitted_module_types;
+  } else {
+    app_params.moduleType->swap(valid_module_types);
+  }
+}
+
 /**
  * @brief FilterPolicyTable filter values that not present in schema
  * @param pt policy table to filter
  */
 void FilterPolicyTable(policy_table::PolicyTable& pt) {
+  policy_table::ModuleConfig& module_config = pt.module_config;
```

Comment:
@JackLivio could be const?

Reply:
Not this one, module_config gets modified